src/sonic-dhcp-server/dhcp_server/dhcprelayd.py

```diff
@@ -0,0 +1,197 @@
+import syslog
+import time
+from swsscommon import swsscommon
+from .dhcp_server_utils import DhcpDbConnector
+
+REDIS_SOCK_PATH = "/var/run/redis/redis.sock"
+DHCP_SERVER_IPV4_SERVER_IP = "DHCP_SERVER_IPV4_SERVER_IP"
+DHCP_SERVER_IPV4 = "DHCP_SERVER_IPV4"
+VLAN = "VLAN"
+DEFAULT_SELECT_TIMEOUT = 5000  # millisecond
+DHCP_SERVER_INTERFACE = "eth0"
+DEFAULT_REFRESH_INTERVAL = 2
+
+KILLED_OLD = 1
+NOT_KILLED = 2
+NOT_FOUND_PROC = 3
+
+
+class DhcpRelayd(object):
+    sel = None
+    subscribe_dhcp_server_table = None
+    subscribe_vlan_table = None
+    dhcp_interfaces_state = {}
+
+    def __init__(self, db_connector, select_timeout=DEFAULT_SELECT_TIMEOUT):
+        """
+        Args:
+            db_connector: db connector obj
+            select_timeout: timeout setting for subscribe db change
+        """
+        self.db_connector = db_connector
+        self.last_refresh_time = None
+        self.select_timeout = select_timeout
+
+    def start(self):
+        """
+        Start function
+        """
+        self.refresh_dhcrelay()
+        self._subscribe_config_db()
+
+    def refresh_dhcrelay(self):
+        """
+        To refresh dhcrelay/dhcpmon process (start or restart)
+        """
+        dhcp_server_ip = self._get_dhcp_server_ip()
+        dhcp_server_ipv4_table = self.db_connector.get_config_db_table(DHCP_SERVER_IPV4)
+        vlan_table = self.db_connector.get_config_db_table(VLAN)
+
+        dhcp_interfaces = set()
+        self.dhcp_interfaces_state = {}
+        for dhcp_interface, config in dhcp_server_ipv4_table.items():
+            self.dhcp_interfaces_state[dhcp_interface] = config["state"]
+            if dhcp_interface not in vlan_table:
+                continue
+            if config["state"] == "enabled":
+                dhcp_interfaces.add(dhcp_interface)
+        self._start_dhcrelay_process(dhcp_interfaces, dhcp_server_ip)
+        self._start_dhcpmon_process(dhcp_interfaces)
+
+    def wait(self):
+        """
+        Wait function, check db change here
+        """
+        while True:
+            self._config_db_update_event()
+
+    def _subscribe_config_db(self):
+        self.sel = swsscommon.Select()
+        self.subscribe_dhcp_server_table = swsscommon.SubscriberStateTable(self.db_connector.config_db,
+                                                                           DHCP_SERVER_IPV4)
+        self.subscribe_vlan_table = swsscommon.SubscriberStateTable(self.db_connector.config_db, VLAN)
+        # Subscribe dhcp_server_ipv4 and vlan table. No need to subscribe vlan_member table
+        self.sel.addSelectable(self.subscribe_dhcp_server_table)
+        self.sel.addSelectable(self.subscribe_vlan_table)
+
+    def _config_db_update_event(self):
+        state, _ = self.sel.select(self.select_timeout)
+        if state == swsscommon.Select.TIMEOUT or state != swsscommon.Select.OBJECT:
+            return
+
+        self._dhcp_server_update_event()
+        self._vlan_update_event()
```

Comment:
trigger update event after line 88 if any of 87/88 updated

Reply:
Fixed